hamilton/plugins/pandas_extensions.py

```diff
@@ -13,6 +13,20 @@
 except ImportError:
     raise NotImplementedError("Pandas is not installed.")
 
+from pandas._libs import lib
+from pandas._typing import (
+    CompressionOptions,
+    DtypeArg,
+    DtypeBackend,
+    FilePath,
+    JSONEngine,
+    JSONSerializable,
+    ReadBuffer,
+    StorageOptions,
+    TimeUnit,
+    WriteBuffer,
+)
```

Comment:
so we might have to drop some of them, or do something specific based on pandas version. E.g. `TimeUnit` doesn't exist in pandas 2.0.2.

If it's a "literal" we should just make it a string type annotation to get around this.